libavcodec/vaapi_decode.c

```diff
@@ -564,10 +586,32 @@ static int vaapi_decode_make_config(AVCodecContext *avctx,
 
     if (frames_ref) {
         AVHWFramesContext *frames = (AVHWFramesContext *)frames_ref->data;
+        AVVAAPIFramesContext *avfc = frames->hwctx;
 
         frames->format = AV_PIX_FMT_VAAPI;
         frames->width = avctx->coded_width;
         frames->height = avctx->coded_height;
+        avfc->enable_sub_frame = (avctx->export_side_data & AV_CODEC_EXPORT_DATA_SUB_FRAME) && support_sfc;
+
+        if (avfc->enable_sub_frame) {
+            avfc->sub_frame_width = avctx->coded_width;
+            avfc->sub_frame_height = avctx->coded_height;
+            avfc->sub_frame_sw_format = AV_PIX_FMT_NV12;
+            if (avctx->sub_frame_opts) {
+                AVDictionaryEntry *e = NULL;
+                while ((e = av_dict_get(avctx->sub_frame_opts, "", e, AV_DICT_IGNORE_SUFFIX))) {
```

Comment:
how do we support multiple sub frames ?

Reply:
The cmdline can be set as "-sub_frame_opts width=640:height=480:format=nv12:width=320:height=240:format=nv12....".
And here we can loop the options and see how many width/height/format can get. 
AV_DICT_IGNORE_SUFFIX this flag return the first AVDictionaryEntry and then remove it from the list.